components/Collections.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  Button,
+  ButtonGroup,
+  Spinner,
+  Stack,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import { getServicesList } from 'api'
+import { ServicesList } from 'models'
+import { UserDoc } from 'models/users'
+import React, { useEffect, useState } from 'react'
+import CollectionItem from './CollectionItem'
+
+interface CollectionProps {
+  userData: UserDoc
+}
+
+export const Collections = ({ userData }: CollectionProps): JSX.Element => {
+  const [loading, setLoading] = useState(true)
+  const [listData, setListData] = useState<ServicesList[]>([])
+
+  useEffect(() => {
+    const promises = []
+    for (const list of userData.lists) {
+      const listData = getServicesList(list)
+      promises.push(listData)
+    }
+
+    Promise.all(promises.map((p) => p.catch((e) => e))).then((list) => {
+      setListData(list.filter((l) => !(l instanceof Error)))
+      setLoading(false)
+    })
+  }, [userData.lists])
```

Comment:
Use [`useQuery`](https://tanstack.com/query/v3/docs/react/reference/useQuery) for this - the reason we use the TanStack Query library so that the state of data fetched from APIs is all handled by the library so we don't need to set up our own React state for loading and API responses.

Also, this `useEffect` is likely unnecessary because users don't create collections from the profile page, a user's collections should never change while they're on their profile page unless someone is directly modifying Firebase or two people are logged into the same account (does Firebase auth even allow this?) and creating collections.

Reply:
Will refactor to use ``useQuery``. Also, I added that functionality in the ``useEffect`` in case someone had multiple tabs open which is possible. I don't think it hurts to have?

Reply:
The `useQuery` refactor hasn't actually been done and we're still using `useEffect` here.

If the confusion issue is resolving promises for multiple queries, [there's a function for that](https://tanstack.com/query/v3/docs/react/reference/useQueries).

Reply:
Oh my bad I completely forgot about that